carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -197,104 +296,330 @@ std::shared_ptr<Geofence> WMBroadcaster::geofenceFromMsg(const cav_msgs::Traffic
+  {
+    back_llt_diag.addRegulatoryElement(regem);
+  }
 
-  // If the days of the week are specified then convert them to the boost format. 
-  GeofenceSchedule::DayOfTheWeekSet week_day_set = { 0, 1, 2, 3, 4, 5, 6 }; // Default to all days  0==Sun to 6==Sat
-  if (msg_schedule.dow_exists) {
-   // sun (6), mon (5), tue (4), wed (3), thu (2), fri (1), sat (0)
-   week_day_set.clear();
-   for (size_t i = 0; i < msg_schedule.dow.dow.size(); i++) {
-     if (msg_schedule.dow.dow[i] == 1) {
-      switch(i) {
-          case 6: // sun
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Sunday);
-            break;
-          case 5: // mon
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Monday);
-            break;
-          case 4: // tue
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Tuesday);
-            break;
-          case 3: // wed
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Wednesday);
-            break;
-          case 2: // thur
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Thursday);
-            break;
-          case 1: // fri
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Friday);
-            break;
-          case 0: // sat
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Saturday);
-            break;
-          default:
-            throw std::invalid_argument("Unrecognized weekday value: " + std::to_string(i));
-      }
-     }
-   }
+  // create new middle lanelets to match the parallel direction
+  std::vector<lanelet::Lanelet> middle_llts;
+  for (auto llt : *middle_opposite_lanelets.get())
+  {
+    lanelet::Lanelet middle_llt (lanelet::utils::getId(), llt.rightBound3d().invert(), llt.leftBound3d().invert());
+    for (auto regem : llt.regulatoryElements())
+    {
+      middle_llt.addRegulatoryElement(regem); //copy existing regem into the new llts
+    }
+    middle_llts.push_back(middle_llt);
   }
 
-  if (msg_schedule.between_exists) {
+  // add traffic light and stoprule to the lanelet of parallel direction
+  lanelet::LineString3d parallel_stop_ls(lanelet::utils::getId(), {parallel_llt_front.leftBound3d().back(), parallel_llt_front.rightBound3d().back()});
+  lanelet::CarmaTrafficLightPtr tfl_parallel = std::make_shared<lanelet::CarmaTrafficLight>(lanelet::CarmaTrafficLight::buildData(lanelet::utils::getId(), {parallel_stop_ls}, {parallel_llt_front}, {}));
+  gf_ptr->traffic_light_id_lookup_.push_back({generate32BitId(work_zone_geofence_cache[cav_msgs::TrafficControlDetail::TAPERRIGHT]->label_),tfl_parallel->id()});
+  lanelet::StopRulePtr stop_rule_parallel = std::make_shared<lanelet::StopRule>(lanelet::StopRule::buildData(lanelet::utils::getId(), {parallel_stop_ls}, {lanelet::Participants::Vehicle}));
```

Comment:
I'll double check, but I don't think you need these StopRule regulatory elements. They are used when there is no other corresponding indicator, but that role is filled by the traffic light stop line in this case.

Reply:
Yeah you can remove these stop rules

Reply:
I have deleted it.